node/pkg/reporter/reporter.go

```diff
@@ -77,37 +78,37 @@ func (r *Reporter) Run(ctx context.Context) {
 	}
 }
 
-func (r *Reporter) report(ctx context.Context, submissionPairs map[int32]SubmissionPair) error {
+func (r *Reporter) report(ctx context.Context, pairs []string) error {
```

Comment:
how about removing pairs parameter and just accessing it through `r.Pairs`

Reply:
ah that's necessary because the `deviationJob` submits a list of deviating pairs

Reply:
then in terms of readability, let's split report function so that it is referenced from same level of abstraction

so `reportJob` and `report` will do. how do you think? `reportJob` and `deviationJob` can then be read in same level, and `report` can be read in on lower level.

how do you think

Reply:
sounds good, will make an update

Reply:
or `regularReportJob`?

Reply:
hmm actually i'm a bit confused. Could you pls elaborate a bit more on how that design removes `pairs` parameter from the `report` function? `report()` still needs to know which pairs to submit, which keeps changing in case of the deviation reporter/job. Maybe a rough pseudocode? 😄

Reply:
Absolutely, 
```
func NewReporter(..) (..) {

...
if config.JobType == ReportJob {
		reporter.Job = func() error {
			return reporter.regularReportJob(ctx)
		}
	}
...


func (r *Reporter) regularReportJob(ctx context.Context) error {
return r.report(ctx, r.Pairs)
}

func (r *Reporter) report(ctx context.Context, pairs []string) error {
...
}
```

Reply:
oh i see, okay :)